Src/FluentAssertions/Specialized/ActionAssertions.cs

```diff
@@ -167,14 +164,13 @@ public void NotThrowAfter(TimeSpan waitTime, TimeSpan pollInterval, string becau
                 }
 
                 invocationEndTime = watch.Elapsed;
-                Thread.Sleep(pollInterval);
+                Task.Delay(pollInterval).GetAwaiter().GetResult();
```

Comment:
How about copying the behavior from [nunit](https://github.com/nunit/nunit/blob/ef49aa71b937a95394abdbaec91911adb9fd58ad/src/NUnitFramework/framework/Internal/ThreadUtility.cs#L38-L45) to get the best possible from all worlds?

```c#
    internal static void BlockingDelay(int milliseconds)
    {
#if !PARALLEL
        System.Threading.Tasks.Task.Delay(milliseconds).GetAwaiter().GetResult();
#else
        Thread.Sleep(milliseconds);
#endif
    }
```

where `PARALLEL` is [defined](https://github.com/nunit/nunit/blob/ef49aa71b937a95394abdbaec91911adb9fd58ad/src/NUnitFramework/Directory.Build.props#L22-L23) as
```xml
<DefineConstants Condition="'$(TargetFramework)' != 'netstandard1.4'
                            and '$(TargetFramework)' != 'netcoreapp1.1'">$(DefineConstants);PARALLEL;SERIALIZATION</DefineConstants>
```

we could use the `!(NETSTANDARD1_3 || NETSTANDARD1_6)` instead of `PARALLEL`

Reply:
I find this interesting, because as far as I'm aware there is no benefit to using one or the other, considering both are blocking calls anyway and from my understanding offer similar performance. For me it makes sense to just use Task.Delay.

Do you know of a particular reason to favour Thread.Sleep in some situations instead? I don't doubt that this was done for a good reason for NUnit, I just want to make sure I understand the decision. 

Thanks very much :)

Reply:
Wondering the same myself...

Reply:
@dennisdoomen @jnyrup 

I've looked extensively into this and can't find any reason to use Thread.Sleep instead. Unless I'm missing something, are we happy to say just using Task.Delay is fine?

Reply:
I'll be the first to admit that I'm no async expert.
I don't know if there are any differences between the two approaches, which are relevant to us.
E.g. whether `Task.Delay().GetAwaiter().GetResult()` is sync-over-async and could cause a deadlock.

I found three related SO posts, but no indication of whether one should be avoid at all costs.
https://stackoverflow.com/questions/53648014/what-is-going-on-with-task-delay-wait
https://stackoverflow.com/questions/34052381/thread-sleep2500-vs-task-delay2500-wait
https://stackoverflow.com/questions/20082221/when-to-use-task-delay-when-to-use-thread-sleep

To summarize, I'm fine with `Task.Delay`.

Reply:
That top link is actually really interesting. I'll try to benchmark that at some point and figure out if it's a major issue. 

I guess in general we should make sure whenever we're using Thread.Sleep or Task.Delay, we treat it less like "I'm guaranteed to wait **exactly** x milliseconds" and more like "I'm guaranteed to wait **at least** x milliseconds".

Reply:
I'm fine with that.